apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -293,7 +296,46 @@ impl MappedResponse {
                         }
                     };
                 }
-                ResponseKey::BatchEntity { .. } => {} // TODO $batch
+                ResponseKey::BatchEntity { keys, inputs, .. } => {
+                    if let serde_json_bytes::Value::Array(values) = value {
+                        let key_selection = keys
+                            .to_mapping()
+                            .map_err(|e| HandleResponseError::MergeError(e.to_string()))?;
+
+                        // Convert representations into keys for use in the map
+                        let key_values = inputs
+                            .batch
+                            .iter()
+                            .map(|v| {
+                                key_selection
+                                    .apply_to(&Value::Object(v.clone()))
+                                    .0
+                                    .unwrap_or(Value::Null)
+                            })
+                            .collect_vec();
```

Comment:
```suggestion
                            });
```
Don't need to allocate the vec here